opcpa-tpr-config/opcpa-tpr-config.py

```diff
@@ -0,0 +1,164 @@
+import yaml
+
+from functools import partial
+
+from qtpy.QtWidgets import QGridLayout
+
+from ophyd import EpicsSignal
+
+from pydm import Display
+from pydm.widgets import PyDMLabel, PyDMPushButton, PyDMNTTable
+
+
+class App(Display):
+    def __init__(self, parent=None, args=None, macros=None):
+        # print(f'args={args}, macros={macros}')
+
+        # Read in config file
+        # TODO add argparse, make this an arg
+        cfg_file = "opcpa-tpr-config/neh_config.yaml"
+
+        config = {}
+        with open(cfg_file, "r") as f:
+            config = yaml.safe_load(f)
+        self.config = config
+
+        print(f"Loaded config {cfg_file}")
+        # print(config)
+
+        # Call super after handling args/macros but before doing pyqt stuff
+        super().__init__(parent=parent, args=args, macros=macros)
+
+        # Now it is safe to refer to self.ui and access your widget objects
+        # It is too late to do any macros processing
+
+        # Setup main areas of GUI
+        self.setup_main()
+
+        # Setup laser specific portions of GUI
+        for laser in self.config["lasers"]:
+            self.setup_rbvs(laser)
+            self.setup_configs(laser)
+
+    def ui_filename(self):
+        return "opcpa-tpr-config.ui"
+
+    def setup_main(self):
+        """
+        Setup the central widgets for the screen.
+        """
+        title = self.findChild(PyDMLabel, "screen_title")
+        if title is not None:
+            title.setText(self.config["main"]["title"])
+        xpm_label = self.findChild(PyDMLabel, "xpm_label")
+        xpm_pv = self.config["main"]["xpm_pv"]
+        if xpm_label is not None:
+            xpm_label.setText(xpm_pv)
+        xpm = self.findChild(PyDMNTTable, "xpm")
+        if xpm is not None:
+            xpm.set_channel(f"pva://{xpm_pv}")
```

Comment:
There's something fascinating about this pva table, I'm not sure what it is. Something about the way the rates dance up and down maybe.